### DISCUSSION TITLE
Toolchain updates between major NDK releases

### DISCUSSION THREAD

**Discussion by Zingam** [android/ndk#1719] [Q&A] [answered]
> NDK r26 is the LTS release that will ship in 2023. Due to bandwidth constraints there are no major releases planned between r25 and the 2023 LTS. r25 is the 2022 LTS so it will be supported at least until r26 ships.

Since there will be such a huge gap between r25 and r26 will there be any Clang updates between these major updates in the minor releases?

**Comment by DanAlbert** [comment, marked as answer]
No. The toolchain team is the one with the lack of bandwidth.

**Comment by Zingam** [comment]
@DanAlbert If there won't be an update until an year from now. Would it be possible to just take these prebuilts from here: 

https://android.googlesource.com/platform/prebuilts/clang/host/
https://android.googlesource.com/platform/prebuilts/clang/host/windows-x86/+/e747503fe55b73a9a7612119d0a0393849869ac5

and replace the ones from the NDK, would that work?

I mean for development only and not for production. I don't know if its even possible to check out those files.

**Reply by stephenhines** [reply]
Those prebuilts may not be as well-tested on Windows and/or Darwin, since we will do most of the evaluation for our AOSP platform builds (and even that might not catch subtle issues with how the new toolchain interacts with the rest of the NDK). You can certainly try them and feel free to report any issues you encounter, however.

**Reply by jfgoog** [reply]
You can download full dev builds of the NDK from https://ci.android.com/builds/branches/aosp-master-ndk/grid

I believe that is our recommended solution now that we are not doing fewer previews and minor releases.

The builds have decent CI tests for Windows, Darwin, and Linux, and should get reasonably regular toolchain updates. I would recommend trying them out rather than swapping in a prebuilt clang yourself.

**Reply by jfgoog** [reply]
If you need a toolchain more recent than the NDK tip of tree, there are instructions for checking out and building the NDK at https://android.googlesource.com/platform/ndk/+/master/docs/Building.md and https://android.googlesource.com/platform/ndk/+/master/docs/Toolchains.md

**Reply by DanAlbert** [reply]
Those instructions won't help you, because we likely will not be updating the NDK's dev branch with newer toolchains, since we can't get the support needed to stabilize them. Your original link is going to be as good as it gets until the toolchain team has the bandwidth to support more than just the LTS release again, but like @stephenhines said, only the toolchains that are shipped in the NDK actually get any Windows testing.
